doc/pantry.md

```diff
@@ -0,0 +1,482 @@
+because when someone else tries to build the project they can get a
+different checkout of the package.
+
+A common practice in the Haskell world is to use "megarepos", or
+repositories with multiple packages in various subdirectories. Some
+common examples include [wai](https://github.com/yesodweb/wai/) and
+[digestive-functors](https://github.com/jaspervdj/digestive-functors). To
+support this, you may also specify `subdirs` for repositories, e.g.:
+
+```yaml
+extra-deps:
+- git: git@github.com:yesodweb/wai
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+  subdirs:
+  - auto-update
+  - wai
+```
+
+Since v1.7.1, you can specify packages from GitHub repository name using `github`:
+
+```yaml
+extra-deps:
+- github: snoyberg/http-client
+  commit: a5f4f30f01366738f913968163d856366d7e0342
+```
+
+If unspecified, `subdirs` defaults to `['.']` meaning looking for a
+package in the root of the repo.  Note that if you specify a value of
+`subdirs`, then `'.'` is _not_ included by default and needs to be
+explicitly specified if a required package is found in the top-level
+directory of the repository.
+
+Using the `stack freeze` command will add in additional information,
+including not only the Pantry tree hash, but also package metadata
+which can allow Stack to work faster by bypassing cabal file
+parses. For example, this:
+
+```yaml
+extra-deps:
+- git: git@github.com:yesodweb/wai
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+  subdirs:
+  - auto-update
+  - wai
+```
+
+Would be converted into:
+
+```yaml
+extra-deps:
+- subdir: auto-update
+  cabal-file:
+    size: 1219
+    sha256: c07b2b1a2df1199f83eef819ac9bb067567e100b60586a52f8b92fc733ae3a6d
+  name: auto-update
+  version: 0.1.2.1
+  git: git@github.com:yesodweb/wai
+  pantry-tree:
+    size: 687
+    sha256: 26377897f35ccd3890b4405d72523233717afb04d62f2d36031bf6b18dcef74f
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+- subdir: wai
+  cabal-file:
+    size: 1717
+    sha256: 7b46e7a8b121d668351fa8a684810afadf58c39276125098485203ef274fd056
+  name: wai
+  version: 3.0.2.3
+  git: git@github.com:yesodweb/wai
+  pantry-tree:
+    size: 10299
+    sha256: ce33fddab13592c847fbd7acd1859dfcbb9aeb6c212db3cee27c909fa3f3ae44
+  commit: 2f8a8e1b771829f4a8a77c0111352ce45a14c30f
+```
+
+### Archives (HTTP(S) or local filepath)
+
+You can use HTTP and HTTPS URLs and local filepaths referring to
+either tarballs or ZIP files.
+
+__NOTE__ Stack assumes that these files never change after downloading
+to avoid needing to make an HTTP request on each build. Use hashes to
+provide more security.
+
+```yaml
+extra-deps:
+- https://example.com/foo/bar/baz-0.0.2.tar.gz
+- archive: http://github.com/yesodweb/wai/archive/2f8a8e1b771829f4a8a77c0111352ce45a14c30f.zip
+  subdirs:
+  - wai
+  - warp
+- archive: ../acme-missiles-0.3.tar.gz
+  sha256: e563d8b524017a06b32768c4db8eff1f822f3fb22a90320b7e414402647b735b
+```
+
+With the `stack freeze` command, this would be replaced with:
```

Comment:
Likewise, if I update and re-run stack freeze will that overwrite the details, here?